docs/fencing/README.md

```diff
@@ -0,0 +1,41 @@
+
+## Purpose and Motivation
+
+The primary motivation for fencing mechanism is to provide an even more **economical solution for HA deployments at the edge**. Customers at edge locations often require redundancy but also need a low cost option to achieve it. This configuration allows organizations to deploy OpenShift with **only two regular-sized control plane nodes**. This helps maintain HA while keeping compute costs down. Extra components needed are supplied in the form of software. 
+
+## High level overview
+
+The TNF architecture leverages **traditional high-availability concepts and technologies to manage critical OpenShift components**, specifically <u>CRI-O, Kubelet, and Etcd</u>. This is achieved through the use of RHEL-HA components: <u>Pacemaker and Corosync</u>.
+- **Pacemaker** acts as the "brain," providing HA capabilities and managing resources through resource agents and recovery functions through fencing agents. Corosync provides the necessary networking and messaging components.
+  - **Etcd management by Pacemaker** is central to maintaining quorum in a two-node setup. Unlike normal OpenShift clusters where etcd runs as pods within the openshift-etcd namespace, in TNF, **Pacemaker runs etcd using Podman containers outside of the cluster**'s direct management.
+- **Corosync** is a core component of the RHEL-HA stack, working alongside Pacemaker. It is an open-source project that provides essential networking and messaging components for the cluster. 
+  - **Corosync is responsible for providing a consistent view of cluster membership, reliable ordered messaging, and flexible quorum capabilities**. In a two-node TNF cluster, Corosync **ensures that the nodes** can form a "C-quorum" (Corosync quorum) by establishing **membership with each other**
+---
+## Installation and Initialization 
+>Note: This process is handled via the provided [dev deployment guide](../../deploy/ipi-baremetalds-virt/README.md), but is described here for your information
+
+ The installation process starts with an initial boostrap node, which will then reboot into one of two final control plane nodes. A critical step involves the Cluster Etcd Operator (CEO), which, upon detecting the DualReplica control plane topology, triggers a TNF controller job. This controller runs pcs commands to initialize Pacemaker and configure the podman-etcd resource agent on both nodes. CEO then relinquishes control of etcd by setting specific flags, causing etcd containers to be removed from static pod configs and immediately recreated by Pacemaker using Podman. Fencing credentials, collected via the install-config (specifically RedFish details like address, username, and password), are made available to the nodes and used by Pacemaker to set up fencing. The Bare Metal Operator (BMO) is also adjusted to prevent power management of control-plane nodes in this topology to avoid conflicts with Pacemaker.
```

Comment:
Bootstrap into one of the cluster nodes is only part of the assisted installer and agent installer workflows. In 4.19 (and currently in 4.20) we only support baremetal IPI or UPI deployments via the core installers - both of which use a bootstrap to initial 2 other nodes.